### ISSUE TITLE
anyio.to_thread.run_sync gives: Cannot access attribute "run_sync" for class "type[BrokenWorkerInterpreter]"

### ISSUE THREAD

**Issue by chriswithers-fuse** [microsoft/pyright#10912]
**Describe the bug**

Sample code:

```
import anyio

anyio.to_thread.run_sync
``
- 
- Python 3.12.4
- pyright v1.1.405
- anyio v4.10.0

```
% uv run pyright repro.py          
.../repro.py
  ....repro.py:3:23 - error: Cannot access attribute "run_sync" for class "type[BrokenWorkerInterpreter]"
    Attribute "run_sync" is unknown (reportAttributeAccessIssue)
1 error, 0 warnings, 0 informations
```

but:

```
% uv run python repro.py 
<function run_sync at 0x102c12e80>
```

**Comment by erictraut** [comment]
Pyright is working correctly here. The top-level module of `anyio` does not automatically load the submodule `to_thread`, so you need to import it to guarantee that your code will work.

```python
import anyio.to_thread
```

If you open `anyio/__init__.py` and search for `to_thread`, you'll see that the submodule isn't referenced anywhere (at least not directly). It may happen to run because some other submodule imports `to_thread`, but relying on that behavior will lead to fragile code that could break the next time the library is updated.

For more details about import loader side effects, refer to [this documentation](https://microsoft.github.io/pyright/#/import-statements).
